rest/data/admin.go

```diff
@@ -52,25 +53,85 @@ func (ac *DBAdminConnector) SetEvergreenSettings(changes *restModel.APIAdminSett
 	newSettings := i.(evergreen.Settings)
 
 	if persist {
-		return &newSettings, evergreen.UpdateConfig(&newSettings)
+		err = evergreen.UpdateConfig(&newSettings)
+		if err != nil {
+			return nil, errors.Wrap(err, "error saving new settings")
+		}
+		return &newSettings, LogConfigChanges(&newSettings, oldSettings, u)
 	}
 
 	return &newSettings, nil
 }
 
-// SetAdminBanner sets the admin banner in the DB and event logs it
-func (ac *DBAdminConnector) SetAdminBanner(text string, u *user.DBUser) error {
-	oldSettings, err := evergreen.GetConfig()
+func LogConfigChanges(newSettings *evergreen.Settings, oldSettings *evergreen.Settings, u *user.DBUser) error {
+	// log the root config document here
+	catcher := grip.NewSimpleCatcher()
+	changes, err := newSettings.GetChanges(oldSettings)
 	if err != nil {
-		return err
+		catcher.Add(errors.Wrap(err, "error getting changes for root document"))
 	}
+	if err := event.LogAdminEvent(newSettings.SectionId(), changes, u.Username()); err != nil {
+		catcher.Add(errors.Wrap(err, "error saving event log for root document"))
+	}
+
+	// log the other config sub-documents
+	valConfig := reflect.ValueOf(*newSettings)
+	var oldStruct reflect.Value
+	if oldSettings != nil {
+		oldStruct = reflect.ValueOf(*oldSettings)
+	}
+
+	//iterate over each field in the config struct
+	for i := 0; i < valConfig.NumField(); i++ {
+		// retrieve the 'id' struct tag
+		sectionId := valConfig.Type().Field(i).Tag.Get("id")
+		if sectionId == "" { // no 'id' tag means this is a simple field that we can skip
+			continue
+		}
+
+		// get the property name and find its value within the settings struct
+		propName := valConfig.Type().Field(i).Name
+		propVal := valConfig.FieldByName(propName)
+
+		// create a reflective copy of the struct
+		valPointer := reflect.Indirect(reflect.New(propVal.Type()))
+		valPointer.Set(propVal)
+
+		// convert the pointer to that struct to an empty interface
+		propInterface := valPointer.Addr().Interface()
+
+		// type assert to the ConfigSection interface
+		section, ok := propInterface.(evergreen.ConfigSection)
+		if !ok {
+			catcher.Add(fmt.Errorf("unable to convert config section %s", propName))
+			continue
+		}
 
-	err = evergreen.SetBanner(text)
+		// save changes in the event log
+		if oldSettings != nil {
+			oldVal := oldStruct.FieldByName(propName)
+			changes, err = section.GetChanges(oldVal.Interface())
```

Comment:
related to our discussion earlier, is there a risk of logging sensitive info the event log?